src/Features/Core/Portable/IntroduceVariable/AbstractIntroduceParameterService.cs

```diff
@@ -0,0 +1,951 @@
+            var options = await document.GetOptionsAsync(cancellationToken).ConfigureAwait(false);
+            var newMethodDeclaration = codeGenerationService.CreateMethodDeclaration(newMethod, options: new CodeGenerationOptions(options: options, parseOptions: expression.SyntaxTree.Options));
+            return newMethodDeclaration;
+        }
+
+        /// <summary>
+        /// Generates a method declaration containing a call to the method that introduced the parameter.
+        /// Example:
+        /// 
+        /// ***This is an intermediary step in which the original function has not be updated yet
+        /// public void M(int x, int y)
+        /// {
+        ///     int f = [|x * y|];
+        /// }
+        /// 
+        /// ---------------------------------------------------->
+        /// 
+        /// public void M(int x, int y) // Generated overload
+        /// {
+        ///     M(x, y, x * y);
+        /// }
+        /// 
+        /// public void M(int x, int y) // Original function
+        /// {
+        ///     int f = x * y;
+        /// }
+        /// </summary>
+        private static async Task<SyntaxNode> GenerateNewMethodOverloadAsync(Document document, TExpressionSyntax expression,
+            IMethodSymbol methodSymbol, int insertionIndex, SyntaxGenerator generator, CancellationToken cancellationToken)
+        {
+            var arguments = generator.CreateArguments(methodSymbol.Parameters);
+
+            // Remove trivia so the expression is in a single line and does not affect the spacing of the following line
+            arguments = arguments.Insert(insertionIndex, generator.Argument(expression.WithoutTrivia()));
+            var memberName = methodSymbol.IsGenericMethod
+                ? generator.GenericName(methodSymbol.Name, methodSymbol.TypeArguments)
+                : generator.IdentifierName(methodSymbol.Name);
+            var invocation = generator.InvocationExpression(memberName, arguments);
+
+            SyntaxNode newStatement;
+            if (!methodSymbol.ReturnsVoid)
+            {
+                newStatement = generator.ReturnStatement(invocation);
+            }
+            else
+            {
+                newStatement = generator.ExpressionStatement(invocation);
+            }
+
+            var codeGenerationService = document.GetRequiredLanguageService<ICodeGenerationService>();
+            var options = await document.GetOptionsAsync(cancellationToken).ConfigureAwait(false);
+            var newMethod = CodeGenerationSymbolFactory.CreateMethodSymbol(methodSymbol, statements: SpecializedCollections.SingletonEnumerable(newStatement).ToImmutableArray(), containingType: methodSymbol.ContainingType);
+            var newMethodDeclaration = codeGenerationService.CreateMethodDeclaration(newMethod, options: new CodeGenerationOptions(options: options, parseOptions: expression.SyntaxTree.Options));
+            return newMethodDeclaration;
```

Comment:
extract this piece into a common helper that generates a method with one statement in it.  you have this pattern in the previous method as well.